server/metrics/prometheus.go

```diff
@@ -15,14 +15,44 @@
 package metrics
 
 import (
-	"github.com/gin-gonic/gin"
+	"errors"
+  "fmt"
 
+	"github.com/drone/drone/server"
+	"github.com/gin-gonic/gin"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
+var (
+	// errInvalidToken is returned when the api request token is invalid.
+	errInvalidToken = errors.New("Invalid or missing token")
+)
+
 // PromHandler will pass the call from /api/metrics/prometheus to prometheus
 func PromHandler() gin.HandlerFunc {
+	handler := promhttp.Handler()
+
 	return func(c *gin.Context) {
-		promhttp.Handler().ServeHTTP(c.Writer, c.Request)
+		token := server.Config.Prometheus.Token
+		bearer := fmt.Sprintf("Bearer %s", token)
```

Comment:
Consider instantiating this right before we need this. Right now we don’t really need it if token is the empty string or the auth header isn’t set

Reply:
Since this is just some static assignment I don't believe that you can measure any difference, but yeah...

Reply:
It's not so much a matter of measuring difference, it's more about code readability. It should probably move down closer to where it's being used, since it's only being used once.